aws-glue-datacatalog-client-common/src/main/java/com/amazonaws/glue/catalog/metastore/GlueMetastoreClientDelegate.java

```diff
@@ -200,6 +202,13 @@ public GlueMetastoreClientDelegate(HiveConf conf, AWSGlue glueClient, Warehouse
     this.conf = conf;
     this.glueClient = glueClient;
     this.wh = wh;
+    ExecutorService customExecutor = ThreadExecutorFactory.getCustomThreadPool(conf, NUM_EXECUTOR_THREADS, threadFactory);
+    if (customExecutor != null ) {
+      this.executorService = customExecutor;
+    } else {
+      this.executorService = GLUE_METASTORE_DELEGATE_THREAD_POOL;
```

Comment:
Does this always get created even if we use a customExecutor? This seems like it could be confusing for people making changes in the future - they might not be sure whether to use `this.executorService` or `GLUE_METASTORE_DELEGATE_THREAD_POOL`.

Reply:
Have changed the interface.  Let me know what you think.